pkg/util/helper/workstatus_test.go

```diff
@@ -17,18 +17,689 @@ limitations under the License.
+							Name:      "test-deployment",
+							Namespace: "default",
+						},
+						Status: &runtime.RawExtension{Raw: []byte(`{"replicas": 3}`)},
+						Health: "Healthy",
+					},
+				},
+			}
+		}
+		return work
+	}
+
+	tests := []struct {
+		name            string
+		bindingID       string
+		binding         *workv1alpha2.ResourceBinding
+		works           []*workv1alpha1.Work
+		expectedError   bool
+		expectedStatus  metav1.ConditionStatus
+		expectedApplied bool
+		expectedEvent   string
+	}{
+		{
+			name:      "successful single work aggregation",
+			bindingID: "test-id-1",
+			binding:   createBinding("binding-1", "test-id-1", []string{"member1"}),
+			works: []*workv1alpha1.Work{
+				createWork("work-1", "karmada-es-member1", "test-id-1", true, true),
+			},
+			expectedError:   false,
+			expectedStatus:  metav1.ConditionTrue,
+			expectedApplied: true,
+			expectedEvent:   "Update ResourceBinding(default/binding-1) with AggregatedStatus successfully",
+		},
+		{
+			name:            "work not found",
+			bindingID:       "test-id-2",
+			binding:         createBinding("binding-2", "test-id-2", []string{"member1"}),
+			works:           []*workv1alpha1.Work{},
+			expectedError:   false,
+			expectedStatus:  metav1.ConditionFalse,
+			expectedApplied: false,
+		},
+		{
+			name:      "work not applied",
+			bindingID: "test-id-3",
+			binding:   createBinding("binding-3", "test-id-3", []string{"member1"}),
+			works: []*workv1alpha1.Work{
+				createWork("work-3", "karmada-es-member1", "test-id-3", false, true),
+			},
+			expectedError:   false,
+			expectedStatus:  metav1.ConditionFalse,
+			expectedApplied: false,
+		},
+		{
+			name:      "multiple works for different clusters",
+			bindingID: "test-id-4",
+			binding:   createBinding("binding-4", "test-id-4", []string{"member1", "member2"}),
+			works: []*workv1alpha1.Work{
+				createWork("work-4-1", "karmada-es-member1", "test-id-4", true, true),
+				createWork("work-4-2", "karmada-es-member2", "test-id-4", true, true),
+			},
+			expectedError:   false,
+			expectedStatus:  metav1.ConditionTrue,
+			expectedApplied: true,
+		},
+		{
+			name:      "work without status",
+			bindingID: "test-id-5",
+			binding:   createBinding("binding-5", "test-id-5", []string{"member1"}),
+			works: []*workv1alpha1.Work{
+				createWork("work-5", "karmada-es-member1", "test-id-5", true, false),
+			},
+			expectedError:   false,
+			expectedStatus:  metav1.ConditionFalse,
+			expectedApplied: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			objects := []client.Object{tt.binding}
+			for _, work := range tt.works {
+				objects = append(objects, work)
+			}
+
+			c := fake.NewClientBuilder().
+				WithScheme(scheme).
+				WithObjects(objects...).
+				WithStatusSubresource(tt.binding).
+				Build()
+
+			recorder := record.NewFakeRecorder(10)
+
+			err := AggregateResourceBindingWorkStatus(context.TODO(), c, tt.binding, recorder)
```

Comment:
This test case maybe conflict with #5779

Reply:
We'll update it once #5779 gets approved.

Reply:
I have updated #5779 , thanks for the reminder @XiShanYongYe-Chang